### PR TITLE
Drop support for py3.7

### PR DESCRIPTION
**Pull request by jeremyliweishih**
Unblocks #3275 and #3290.

### CODE REVIEW

**Comment by jeremyliweishih** [comment]
I think this PR might require @dsherry or @chukarsten to remove the 3.7 tests from required checks as well!

File: setup.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
     long_description=long_description,
     long_description_content_type="text/markdown",
     url='https://github.com/alteryx/evalml/',
+    python_requires='>=3.8, <4',
```
Comment (jeremyliweishih) [inline setup.py:21]: Thoughts on this? This is what pip looks at when determining if your python version would work with a library. Saw this i[n the WW repo](https://github.com/alteryx/woodwork/commit/19681ae1190bb7b027d62513137167e61a31b905#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R32).

Comment (freddyaboulton) [reply setup.py:21]: Looks good to me

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @jeremyliweishih ! We'll need @chukarsten to get this in with his magic powers hehe

**Review by chukarsten — APPROVED** [review]

**Pull request merged**